x/ibc/core/spec/01_concepts.md

```diff
@@ -159,23 +266,117 @@ in the handshake callbacks.
+
+A packet may be associated with one of the following states:
+- the packet does not exist (ie it has not been sent)
+- the packet has been sent but not received (the packet commitment exists on the 
+sending chain, but no receipt exists on the receiving chain)
+- the packet has been received but not acknowledged (packet commitment exists 
+on the sending chain, a receipt exists on the receiving chain, but no acknowledgement
+exists on the receiving chain)
+- the packet has been acknowledgement but the acknowledgement has not been relayed 
+(the packet commitment exists on the sending chain, the receipt and acknowledgement
+exist on the receiving chain)
+- the packet has completed its life cycle (the packet commitment does not exist on
+the sending chain, but a receipt and acknowledgement exist on the receiving chain)
+
+Sending of a packet is initiated by a call to the `ChannelKeeper.SendPacket` 
+function by an application module. Packets being sent will be verified for
+correctness (core logic only). If the packet is valid, a hash of the packet
+will be stored as a packet commitment using the packet sequence in the key.
+Packet commitments are stored on the sending chain.
+
+A message should be sent to the receving chain indicating that the packet
+has been committed on the sending chain and should be received on the 
+receiving chain. The light client on the receiving chain, which verifies
+the sending chain's state, should be updated to the lastest sending chain
+state if possible. The verification will fail if the lastest state of the
+light client does not include the packet commitment. The receiving chain
+is responsible for verifying that the counterparty set the hash of the 
+packet. If verification of the packet to be received is successful, the
+receiving chain should store a receipt of the packet and call application
+logic if necessary. An acknowledgement may be stored at this time (synchonously)
+or at another point in the future (aynchronously). 
+
+Acknowledgements written on the receiving chain may be verified on the 
+sending chain. If the sending chain successfully verifies the acknowledgement
+then it may delete the packet commitment stored at that sequence. There is
+no requirement for acknowledgements to be written. Only the hash of the
+acknowledgement is stored on the chain. Application logic may be executed
+in conjunction with verifying an acknowledgement. For example, in fungible
+cross-chain token transfer, a failed acknowledgement results in locked or
+burned funds being refunded. 
+
+Relayers are responsible for reconstructing packets between the sending, 
+receiving, and acknowledging of packets. 
+
+IBC applications sending and receiving packets are expected to appropriately
+handle data contained within a packet. For example, cross-chain token 
+transfers will unmarshal the data into proto definitions representing
+a token transfer. 
+
+Future optimizations may allow for storage cleanup of stored packet 
```

Comment:
Packet commitments are cleaned-up when acknowledgements are processed

Reply:
(you mention this above)

Reply:
reworded